lib/nats/io/client.rb

```diff
@@ -271,6 +273,8 @@ class << self; alias_method :request, :old_request; end
       # Check for TLS usage
       @tls = @options[:tls]
 
+      @inbox_prefix = opts.fetch(:custom_inbox_prefix, @inbox_prefix)
```

Comment:
in go we do a bunch of sanity checks on this, not seeing any kinds of checks for option validity so not sure where to put it :)


https://github.com/nats-io/nats.go/blob/b337a5c01afdab41a22becdf3759e038f9948889/nats.go#L1141

Reply:
I think fine to add here for now

Reply:
I added some setting validation and tests.